### PR TITLE
Argument/Setting to set log path

### PR DESCRIPTION
**Pull request by shantigilbert**
This PR allows to choose the directory where you want to save es_log.txt if none of the options are set it will use the default `$HOME/.emulationstation/es_log.txt`

* Using argument --log-path [path]
* Using es_settings.cfg `<string name="LogPath" value="[path]" />`

As requested on #596

### CODE REVIEW

File: es-core/src/Log.cpp [hunk 1]
```diff
@@ -14,10 +15,14 @@ LogLevel Log::getReportingLevel()
 
 std::string Log::getLogPath()
 {
+	if (Settings::getInstance()->getString("LogPath") != "") {
+	return Settings::getInstance()->getString("LogPath") + "/es_log.txt";
```
Comment (joolswills) [inline es-core/src/Log.cpp]: Indent?

Comment (shantigilbert) [reply es-core/src/Log.cpp]: Would that be better?

**Comment by tomaz82** [comment]
To avoid grabbing the LogPath twice, I personally would've done something like this:
(pseudo code)

`string logpath = Settings::logpath();
if(logpath == "")
	logpath = Utils::home + "/.emulationstation";

return logpath + "/es_log.txt";`

**Comment by tomaz82** [comment]
Also this was not added as a "don't save setting" or defaulted to an empty string.

If this is a requested feature it would probably be better implemented inside Utils::FileSystem the same way homePath and exePath is ( and it would "cache" the path as to not have to reconstruct it every time getLogPath is called )

**Comment by shantigilbert** [comment]
Not sure what you mean by "don't save setting" on my test it did not save the path because I though that is what its supposed to do, it's an temp argument (just like all the othe arguments), but you could also add it manually to es_settings to make it more permanent. 

As far as why it's in Log.ccp and not in FileSystemUltil was because the default path for the log is set there, I figure that was the way it was handled. 

But if what you say is how its supposed to be done I can try and make the changes.

**Comment by tomaz82** [comment]
What I meant was that it was not added to the list of arguments to be excluded when saving es_settings.cfg, so in it's current state my guess is that it does save it on exit.
Also it's never set to a default value anywhere.

**Comment by jrassa** [comment]
https://github.com/RetroPie/EmulationStation/blob/master/es-core/src/Settings.cpp#L15

This what tomaz is refering to.  Generally speaking, a setting that is set via argument shouldn't be preserved in settings.xml.  As is, if you run once and set a custom log path.  The next time you run, even if you don't set a custom log path, the previous path will be used since it will have been saved.  You need to add `LogPath` to `settings_dont_save` to prevent this.

https://github.com/RetroPie/EmulationStation/blob/master/es-core/src/Settings.cpp#L53

Here you should be setting the default value for `LogPath` to ensure that it is initialized properly.

**Comment by shantigilbert** [comment]
@jrassa @tomaz82 Thanks! I added it to the don't save list and the default set to blank, is this better?

**Pull request closed without merging**